eng/Directory.Build.Common.targets

```diff
@@ -12,6 +12,10 @@
     <Error Condition="'@(RequiredTargetFrameworks)' == ''" Text="List of required target frameworks is empty something must have messed up property RequiredTargetFrameworks[$(RequiredTargetFrameworks)]." />
     <Error Condition="'@(MissingTargetFrameworks)' != ''" Text="Missing required target frameworks '@(MissingTargetFrameworks)'. Please ensure you add those frameworks." />
   </Target>
+  
+  <Target Name="ValidateApiCompat" BeforeTargets="Build">
+    <Error Condition="'$(RunApiCompat)' == 'false'" Text="Api Compat cannot be disabled."/>
```

Comment:
Why do we want to enforce this? There are some potential cases were it needs to be disabled, at least temporarily.

Reply:
Can you provide an example of such cases?

Reply:
Ah I think I just hit one - the storage meta package. Do you know of any other examples?

Reply:
> Why do we want to enforce this? 

To prevent shipping breaking changes.

Reply:
There might actually be approved breaking changes that we need to get past. However maybe a baseline file could work for those cases.

Reply:
Yes, I think we use a baseline file for those cases.